pkg/catalogv2/oci/oci_test.go

```diff
@@ -244,3 +246,57 @@ func TestGenerateIndex(t *testing.T) {
 		})
 	}
 }
+
+func TestGenerateIndexExactAndSubMatch(t *testing.T) {
```

Comment:
This is missing the case where `ociClient.repository == ""` e.i. where the user only provides the registry